### PR TITLE
Pushing for help with Redux selectors

### PR DESCRIPTION
**Pull request by daovist**
I watched a Redux video and read some of the docs. I've created my action (please ignore the do and confirm actions, I will delete those, sorry) and reducer and they seem to be working but my selector is not. I could use help on this. I think once I manage to get the videoPause value from the store into the VideoPlayer component I will be able to wrap this up.

### CODE REVIEW

**Comment by liamcardenas** [comment]
I'm looking into this

File: src/renderer/component/video/internal/player.jsx [hunk 1]
```diff
@@ -121,6 +124,11 @@ class VideoPlayer extends React.PureComponent {
     }
   }
 
+  pauseVideo() {
```
Comment (liamcardenas) [inline src/renderer/component/video/internal/player.jsx]: what if you change this to `componentWillReceiveProps` instead of `pauseVideo`?

Comment (liamcardenas) [reply src/renderer/component/video/internal/player.jsx]: then of course you would need to filter out the prop changes that don't lead to a pause, but does that make sense?

Comment (daovist) [reply src/renderer/component/video/internal/player.jsx]: I tried that and the method(?) was never called.

I think the issue is that the selector isn't working, or that the reducer isn't actually changing the state. My focus is on line 63 of components/video.view.jsx where it console logs the value of videoPause. It's always false, even after I've clicked open, which should update valuePause to true through the reducer.

Comment (kauffj) [reply src/renderer/component/video/internal/player.jsx]: @daovist try using the console to help debug what unexpected behavior is happening here. The console will show all action dispatches as well as all state changes (expand the console messages that are printed on action dispatches). This should tell you whether it's an issue with actions or the reducer. If both the action is fired and the Redux state looks correct, you can add a console statement to see if the selector is firing when you think it ought to.

File: src/renderer/component/video/internal/player.jsx [hunk 2]
```diff
@@ -157,28 +166,39 @@ class VideoPlayer extends React.PureComponent {
...
+          )}
+          <div ref="media" className="media" />
+        </div>
       </div>
     );
   }
 }
 
-export default VideoPlayer;
+const select = (state, props) => ({
```
Comment (liamcardenas) [inline src/renderer/component/video/internal/player.jsx]: this should go in videos/index.js

File: src/renderer/component/video/view.jsx [hunk 3]
```diff
@@ -56,8 +57,11 @@ class Video extends React.PureComponent {
       changeVolume,
       volume,
       uri,
+      // videoPause,
```
Comment (liamcardenas) [inline src/renderer/component/video/view.jsx]: you want this in here, and then you want to pass it into VideoPlayer

File: src/renderer/redux/selectors/video.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import * as settings from "constants/settings";
+import { createSelector } from "reselect";
+
+const _selectState = state => state.videoPause || false;
```
Comment (liamcardenas) [inline src/renderer/redux/selectors/video.js]: this should be

`const _selectState = state => state.video || {};`

File: src/renderer/component/video/internal/player.jsx [hunk 5]
```diff
@@ -20,6 +20,13 @@ class VideoPlayer extends React.PureComponent {
     this.togglePlayListener = this.togglePlay.bind(this);
   }
 
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.videoPause) {
+      this.refs.media.children[0].pause();
+      this.props.setVideoPause(false);
```
Comment (liamcardenas) [inline src/renderer/component/video/internal/player.jsx]: is this (`this.props.setVideoPause(false);`) necessary? can we trigger this, instead, when they push "play" in app?

**Comment by liamcardenas** [comment]
Making a follow up issue to this since this is work.

**Pull request merged**